libraries/chain/database.cpp

```diff
@@ -2419,22 +2408,15 @@ namespace golos { namespace chain {
 
                 push_virtual_operation(comment_payout_update_operation(comment.author, to_string(comment.permlink)));
 
-                const auto &vote_idx = get_index<comment_vote_index>().indices().get<by_comment_voter>();
+                const auto& vote_idx = get_index<comment_vote_index>().indices().get<by_comment_voter>();
                 auto vote_itr = vote_idx.lower_bound(comment.id);
-                while (vote_itr != vote_idx.end() &&
-                       vote_itr->comment == comment.id) {
-                    const auto &cur_vote = *vote_itr;
+                while (vote_itr != vote_idx.end() && vote_itr->comment == comment.id) {
+                    const auto& cur_vote = *vote_itr;
                     ++vote_itr;
-                    if (!has_hardfork(STEEMIT_HARDFORK_0_12__177) ||
-                        calculate_discussion_payout_time(comment) !=
-                        fc::time_point_sec::maximum()) {
-                        modify(cur_vote, [&](comment_vote_object &cvo) {
-                            cvo.num_changes = -1;
+                    if (comment.mode == archived) {
```

Comment:
Moving `if` out of scope `while` can increase performance.

Reply:
You're right, thx, fixed